### PR TITLE
common/options: update mon_crush_min_required_version=hammer

### PR DESCRIPTION
**Pull request by liewegas**
- change min to hammer
- update nautilus upgrade docs to migrate to all buckets to straw2

The intention here is to backport this to nautilus for 14.2.2.  I think the timing is right, notwithstanding the fact that we didn't do this for 14.2.0.  We could wait for Octopus instead...

### CODE REVIEW

File: doc/releases/nautilus.rst [hunk 1]
```diff
@@ -319,6 +319,27 @@ Instructions
...
+   If your CRUSH tunables are older than Hammer, Ceph will now issue a
+   health warning.  If you see a health alert to that effect, you can
+   revert this change with::
+
+     ceph config set mon mon_crush_min_required_version firefly
+
+   If Ceph does not complain, however, then we recommend you also
+   switch any existing CRUSH buckets to straw2, which was added back
+   in the Hammer release.  If you have any 'straw' buckets, this will
+   result in a modest amount of data movement, but generally nothing to severe.::
```
Comment (gregsfortytwo) [inline doc/releases/nautilus.rst]: s/to/too

**Comment by gregsfortytwo** [comment]
So chooseleaf_stable came in for Jewel — if we do decide to backport this to Nautilus, should we also "require" jewel tunables for Octopus, or do you have a plan there?

**Comment by liewegas** [comment]
for reference: https://github.com/ceph/ceph/blob/master/src/crush/CrushWrapper.h#L299

**Comment by liewegas** [comment]
> So chooseleaf_stable came in for Jewel — if we do decide to backport this to Nautilus, should we also "require" jewel tunables for Octopus, or do you have a plan there?

I think probably not. I'm not sure if it ever is worth making the hammer -> jewel leap.  IIRC it moves most of the data, and although jewel is better, hammer really isn't that bad.

**Comment by gregsfortytwo** [comment]
> > So chooseleaf_stable came in for Jewel — if we do decide to backport this to Nautilus, should we also "require" jewel tunables for Octopus, or do you have a plan there?
> 
> I think probably not. I'm not sure if it ever is worth making the hammer -> jewel leap. IIRC it moves most of the data, and although jewel is better, hammer really isn't that bad.

It's definitely quoted as moving data, but I don't know what it fixed. I thought it was a pretty serious balance problem that might have a bearing on the balancer's performance? Or maybe it was only when out'ing OSDs that it mattered.

**Pull request merged**